cdc_admin/cdc_admin.thrift

```diff
@@ -0,0 +1,127 @@
+# Copyright 2016 Pinterest Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+namespace cpp2 cdc_admin
+namespace java com.pinterest.cdc_admin.thrift
+
+enum CDCAdminErrorCode {
+  NOT_FOUND = 1,
+  ALREADY_EXIST = 2,
+  INVALID_ROLE = 3,
+  INVALID_UPSTREAM = 4,
+  ADMIN_ERROR = 5,
+}
+
+exception CDCAdminException {
+  1: required string message,
+  2: required CDCAdminErrorCode errorCode,
+}
+
+struct AddObserverRequest {
+  # the db to add an observer for.
+  # if this db is already being observed, an ALREADY_EXIST error is thrown
+  1: required string db_name,
+  2: required string upstream_ip,
+  # add the observer to the db_manager with the specified role. one of LEADER, STANDBY
+  3: optional string db_role = "STANDBY",
+}
+
+struct AddObserverResponse {
+  # for future use
+}
+
+struct RemoveObserverRequest {
+  # the db to remove the observer for
+  1: required string db_name,
+}
+
+struct RemoveObserverResponse {
+  # for future use
+}
+
+struct CheckObserverRequest {
+  # the DB to check for an observer for
+  1: required string db_name,
+}
+
+struct CheckObserverResponse {
+  # the largest sequence number that has been published for this segment
+  1: optional i64 seq_num = 0,
+  # if the observer is Leader or Standby
+  2: optional bool is_leader = false,
+}
+
+struct ChangeObserverRoleAndUpstreamRequest {
+  # the db to change the observer for
+  1: required string db_name,
+  # the role to change to, only "LEADER" and "STANDBY" are supported
+  2: required string new_role,
+  # the new upstream to pull updates from
+  3: optional string upstream_ip,
+  4: optional i16 upstream_port,
+}
```

Comment:
Same comment here. Is the "Role" field applicable?
On becoming a leader, a partition will start observing. On becoming standby, it does nothing.